File: April 11, 2026.js
function rookBishopAttack(rook, bishop) {

  const letterRook = rook[0];
  const letterBishop = bishop[0];
  const nRook = Number(rook.slice(1));
  const nBishop = Number(bishop.slice(1));
  const colRook = letterRook.charCodeAt(0);
  const colBishop = letterBishop.charCodeAt(0);

  if (letterRook == letterBishop || nRook == nBishop) {
    return "rook";
  }

  if (Math.abs(colRook - colBishop) == Math.abs(nRook - nBishop)) {
    return "bishop";
  }
  
  return "neither";
}

rookBishopAttack("G3", "E8")

File: April 11, 2026.py
def rook_bishop_attack(rook, bishop):

    letter_rook = rook[0]
    letter_bishop = bishop[0]
    n_rook = int(rook[1:])
    n_bishop = int(bishop[1:])
    col_rook = ord(letter_rook)
    col_bishop = ord(letter_bishop)

    if letter_rook == letter_bishop or n_rook == n_bishop:
        return "rook"

    if abs(n_rook - n_bishop) == abs(col_rook - col_bishop):
        return "bishop"

    return "neither"